scripts/plotly_streamlit.py

```diff
@@ -181,25 +180,21 @@ def time_resample(df, resample_time):
 font_size = 15
 
 
-def sunrise_sunset_scatter(num_days_to_display):
-    latitude = df2['Lat'][0]
-    longitude = df2['Lon'][0]
+def sunrise_sunset_scatter(date_range):
+    latitude = df2['Lat'][-1]
```

Comment:
Not strictly necessary but I had a secondary problem that others might have: My initial lat/long configuration was wrong. Why does this matter? Well, the existing code uses the earliest lat/long it can find in the database! I cleaned up my data a bit, but decided to include this change.

This is a simple change that uses the *latest* lat/long in the database. I figure that's going to be more correct. Ideally, this would do an actual configuration lookup instead of inferring from db detection records.

Reply:
Well-done! That is a good find! There's been quite a lot of people over the years who didn't initially set their lat/lon and did it afterwards. That would indeed result in the problem. I see what you mean about config lookup too. I agree that it would be ideal. Never looked at the code in Species Stats. Might have to familiarise myself with that!